packages/flutter_tools/test/general.shard/custom_devices/custom_device_test.dart

```diff
@@ -0,0 +1,544 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'dart:async';
+
+import 'package:file/src/interface/directory.dart';
+import 'package:file/src/interface/file.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/build_info.dart';
+import 'package:flutter_tools/src/bundle.dart';
+import 'package:flutter_tools/src/convert.dart';
+import 'package:flutter_tools/src/custom_devices/custom_device.dart';
+import 'package:flutter_tools/src/custom_devices/custom_device_config.dart';
+import 'package:flutter_tools/src/custom_devices/custom_devices_config.dart';
+import 'package:flutter_tools/src/device.dart';
+import 'package:flutter_tools/src/linux/application_package.dart';
+import 'package:flutter_tools/src/globals.dart' as globals;
+
+import 'package:file/memory.dart';
+import 'package:file/file.dart';
+import 'package:flutter_tools/src/project.dart';
+import 'package:mockito/mockito.dart';
+
+import '../../src/common.dart';
+import '../../src/context.dart';
+import '../../src/fake_process_manager.dart';
+import '../../src/fakes.dart';
+
+class MockBundleBuilder extends Mock implements BundleBuilder {}
```

Comment:
Prefer using classes that extends `Fake` instead of adding more mocks. We're trying to avoid using new Mock classes to keep more null safety migration work from piling up.